processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java

```diff
@@ -197,8 +197,9 @@ public AbstractFactDataWriter(String storeLocation, int measureCount, int mdKeyL
     blockIndexInfoList = new ArrayList<>();
     // get max file size;
     CarbonProperties propInstance = CarbonProperties.getInstance();
-    this.fileSizeInBytes = blocksize * CarbonCommonConstants.BYTE_TO_KB_CONVERSION_FACTOR
-        * CarbonCommonConstants.BYTE_TO_KB_CONVERSION_FACTOR * 1L;
+    // if blocksize=2048, then 2048*1024*1024 will beyond the range of Int
+    this.fileSizeInBytes = 1L * blocksize * CarbonCommonConstants.BYTE_TO_KB_CONVERSION_FACTOR
```

Comment:
instead of multiple by `1L`, you can just convert `blocksize` to `(long)blocksize` instead